src/sidebar/build-thread.js

```diff
@@ -114,49 +113,44 @@ function setParentID(threads, id, parents) {
  * @return {Thread} - The input annotations threaded into a tree structure.
  */
 function threadAnnotations(annotations) {
-  // Map of annotation ID -> container
+  /** @type {Object.<string,Thread>} */
   const threads = {};
 
-  // Build mapping of annotation ID -> thread
-  annotations.forEach(function (annotation) {
-    threads[id(annotation)] = Object.assign({}, DEFAULT_THREAD_STATE, {
-      id: id(annotation),
-      annotation: annotation,
-      children: [],
-    });
+  // Create a `Thread` for each annotation
+  annotations.forEach(annotation => {
+    const id = annotationId(annotation);
+    threads[id] = {
+      ...DEFAULT_THREAD_STATE,
+      ...{ children: [], annotation, id },
```

Comment:
You don't need to use object spread here, since you're just adding a static set of properties to an object literal. IOW - `{ ...DEFAULT_THREAD_STATE, children: [], annotation, id }` would be more efficient.

Reply:
My brain must have been on some sort of special "definitely spread everything" trip.